sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_workspace_utils.py

```diff
@@ -84,6 +84,10 @@ def get_resource_and_group_name(armstr: str) -> str:
     return armstr.split("/")[-1], armstr.split("/")[-5]
 
 
+def get_sub_id_resource_and_group_name(armstr: str) -> str:
+    return armstr.split("/")[-7], armstr.split("/")[-1], armstr.split("/")[-5]
```

Comment:
Overall, the change looks good to me. The splits on ARM IDs are often a yellow flag, though. We have had past incidents where some variations of ARM ID would blow off the code. 
I trust you to also analyze this from that perspective!

Reply:
Yes, I reused the code from above